storage_service/administration/forms.py

```diff
@@ -222,10 +222,30 @@ class UserChangeForm(auth.forms.UserChangeForm):
     """ Modifys an existing user.  Inherits from django's UserChangeForm. """
 
     def __init__(self, *args, **kwargs):
+        current_user = kwargs.pop("current_user", None)
+        user_being_edited = kwargs["instance"]
+        self.superusers = auth.get_user_model().objects.filter(is_superuser=True)
         super(UserChangeForm, self).__init__(*args, **kwargs)
         self.fields["is_superuser"].label = _("Administrator?")
+        if not (current_user and current_user.is_superuser):
+            # If current user is not super, do not permit editing of that.
+            del self.fields["is_superuser"]
+        elif self.superusers.count() == 1 and not (current_user != user_being_edited):
+            # Provide some indication that this is undesirable."""
+            self.fields["is_superuser"].widget.attrs["readonly"] = True
         del self.fields["password"]
 
+    def clean(self):
+        """Validate the form to protect against potential user errors."""
+        if self.superusers.count() > 1:
+            return self.cleaned_data
+        try:
+            # Protect field from being reverted if only one superuser.
+            self.cleaned_data["is_superuser"] = True
+        except KeyError:
+            """Field isn't being modified, nothing to do."""
```

Comment:
It's v minor but I would say this is a place for a pass and a comment. Strings are code, even if they are no-ops, comments aren't (docstrings are an exception). [This SO answer is pretty good on that topic](https://stackoverflow.com/questions/708649/python-comments-vs-strings#answer-708674).